crates/polars-plan/src/dsl/function_expr/datetime.rs

```diff
@@ -271,22 +271,13 @@ pub(super) fn date(s: &Series) -> PolarsResult<Series> {
 pub(super) fn datetime(s: &Series) -> PolarsResult<Series> {
     match s.dtype() {
         #[cfg(feature = "timezones")]
-        DataType::Datetime(tu, Some(tz)) => {
-            let mut out = {
-                polars_ops::chunked_array::replace_time_zone(
-                    s.datetime().unwrap(),
-                    None,
-                    &StringChunked::from_iter(std::iter::once("raise")),
-                    NonExistent::Raise,
-                )?
-                .cast(&DataType::Datetime(*tu, None))?
-            };
-            if tz != "UTC" {
-                // DST transitions may not preserve sortedness.
-                out.set_sorted_flag(IsSorted::Not);
-            }
-            Ok(out)
-        },
+        DataType::Datetime(tu, Some(_)) => polars_ops::chunked_array::replace_time_zone(
+            s.datetime().unwrap(),
+            None,
+            &StringChunked::from_iter(std::iter::once("raise")),
+            NonExistent::Raise,
+        )?
+        .cast(&DataType::Datetime(*tu, None)),
```

Comment:
`polars_ops::chunked_array::replace_time_zone` already deals with unsetting the sortedness flag if necessary, so this can now be simplified